pkg/controller/pxc/controller.go

```diff
@@ -78,12 +83,52 @@ func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
 func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	if err := setupFieldIndexers(mgr); err != nil {
+		return errors.Wrap(err, "setup field indexers")
+	}
 	return builder.ControllerManagedBy(mgr).
 		Named(naming.OperatorController).
 		Watches(&api.PerconaXtraDBCluster{}, &handler.EnqueueRequestForObject{}).
+		Watches(&corev1.Secret{}, enqueuePXCReferencingSecret(mgr.GetClient())).
 		Complete(r)
 }
 
+func setupFieldIndexers(mgr manager.Manager) error {
+	return mgr.GetFieldIndexer().IndexField(context.TODO(), &api.PerconaXtraDBCluster{}, secretsNameField, func(o client.Object) []string {
+		cluster, ok := o.(*api.PerconaXtraDBCluster)
+		if !ok {
+			return nil
+		}
+		return []string{cluster.Spec.SecretsName}
+	})
+}
+
+// enqueuePXCReferencingSecret returns an EventHandler that returns a list of all
+// pxc-clusters that reference the secret via `.spec.secretsName`.
+func enqueuePXCReferencingSecret(c client.Client) handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, o client.Object) []reconcile.Request {
+		secret, ok := o.(*corev1.Secret)
+		if !ok {
+			return nil
+		}
+		list := &api.PerconaXtraDBClusterList{}
+		_ = c.List(context.TODO(), list, &client.ListOptions{
```

Comment:
Maybe I'm missing something, but do we have a reason not to handle the error when listing objects? We can error log it then return nil since we have already the ctx available.

Reply:
updated